docs/blog/posts/2024-11-14-pypi-now-supports-digital-attestations.md

```diff
@@ -0,0 +1,116 @@
+---
+title: PyPI now supports digital attestations
+description: Announcing support for PEP 740 on the Python Package Index
+authors:
+  - di
+date: 2024-11-14
+tags:
+  - publishing
+  - security
+  - oidc
+---
+
+PyPI package maintainers can now publish signed digital attestations when
+publishing, in order to further increase trust in the supply-chain security of
+their projects. Additionally, a new API is available for consumers and
+installers to verify published attestations.
+
+Many projects have already begun publishing attestations, with more than 10,000
+attestations already published.
+
+This finalizes PyPI's support for [PEP 740], and follows directly from previous
+work to add support for [Trusted Publishing], as well as the [deprecation and
+removal of PGP signatures].
+
+<!-- more -->
+
+### Why not plain signatures?
+
+PyPI's support for digital attestations has three key advantages over regular
+cryptographic signatures, such as those provided by PGP:
+
+* **Attestations are signed by an identity, not a key pair:**
+  Similar to our recent support for Trusted Publishing, PyPI's support for
+  digital attestations relies upon Open ID Connect (OIDC) identities. By signing
+  attestations with identities, and not a public/private key pair, we mitigate the
+  potential for an individual's key loss or compromise, one of the most common failure cases for
+  PGP signing.
+* **Attestations provide a verifiable link to an upstream source repository:**
+  By signing with the identity of the upstream source repository, such as in the
+  case of an upload of a project built with GitHub Actions, PyPI's support for
+  digital attestations defines a strong and verifiable association between a file
+  on PyPI and the source repository, workflow, and even the commit hash that
+  produced and uploaded the file.
+* **Attestations are verified when uploaded, and must be verifiable to be uploaded:**
+  Upon review of the state of historical PGP signatures published to PyPI, [we
+  found that many signatures were not verifiable], either by PyPI or by end
+  users. With support for PEP 740, PyPI only permits attestations with a
+  verifiable signature to be uploaded and redistributed by the index. This
+  ensures that all attestations are verifiable and useful for all PyPI users.
```

Comment:
Could also mention the transparency and auditability of attestations to mitigate against compromise of the repository?